test/unit/test_helper_test.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module JobIteration
+  class TestHelperTest < ActiveSupport::TestCase
+    include JobIteration::TestHelper
+
+    class CounterJob < ActiveJob::Base
+      include JobIteration::Iteration
+
+      class << self
+        attr_writer :count
+
+        def count
+          @count ||= 0
+        end
+      end
+
+      def build_enumerator(times, cursor:)
+        enumerator_builder.array(Array.new(times), cursor: cursor)
+      end
+
+      def each_iteration(_, _)
+        self.class.count += 1
+      end
+    end
+
+    teardown do
+      CounterJob.count = nil
+      ActiveJob::Base.queue_adapter.enqueued_jobs = []
+    end
+
+    test "#iterate_exact_times interrupts jobs after the given number of iterations" do
+      iterate_exact_times(3.times)
+
+      CounterJob.perform_now(10)
+
+      assert_equal 3, CounterJob.count
+    end
+
+    test "#iterate_once interrupts jobs after a single iteration" do
+      iterate_once
+
+      CounterJob.perform_now(10)
+
+      assert_equal 1, CounterJob.count
+    end
+
+    test "#continue_iterating allows jobs to iterate until the end" do
+      iterate_exact_times(3.times)
+      continue_iterating
+
+      CounterJob.perform_now(10)
+
+      assert_equal 10, CounterJob.count
+    end
+
```

Comment:
There are two extra cases I'd like to add, based on the test that failed in core:

```suggestion
    test "#iterate_once allows jobs to run one iteration at a time" do
      iterate_once

      job = CounterJob.perform_now(10)
      job.perform_now

      assert_equal 2, CounterJob.count
    end

    test "#continue_iterating allows the job to finish after running initial iterations" do
      iterate_once

      job = CounterJob.perform_now(10)

      continue_iterating

      job.perfom_now

      assert_equal 10, CounterJob.count
    end
```